packages/postgres-database/src/simcore_postgres_database/migration/versions/e8057a4a7bb0_new_services_tags_table.py

```diff
@@ -0,0 +1,44 @@
+"""new services_tags table
+
+Revision ID: e8057a4a7bb0
+Revises: 7604e65e2f83
+Create Date: 2024-08-23 12:12:32.883771+00:00
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "e8057a4a7bb0"
+down_revision = "7604e65e2f83"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "services_tags",
+        sa.Column("service_key", sa.String(), nullable=False),
+        sa.Column("service_version", sa.String(), nullable=False),
+        sa.Column("tag_id", sa.BigInteger(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["service_key", "service_version"],
+            ["services_meta_data.key", "services_meta_data.version"],
+            onupdate="CASCADE",
+            ondelete="CASCADE",
+        ),
+        sa.ForeignKeyConstraint(
+            ["tag_id"], ["tags.id"], onupdate="CASCADE", ondelete="CASCADE"
+        ),
+        sa.UniqueConstraint(
+            "service_key", "service_version", "tag_id", name="services_tags_uc"
```

Comment:
Question: what is the advantage of defining a unique constraint instead of a primary key directly?

Reply:
wrote above